sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -337,27 +338,33 @@ public ConfigurationClientBuilder clientOptions(ClientOptions clientOptions) {
      * @throws NullPointerException If {@code connectionString} is null.
      * @throws IllegalArgumentException If {@code connectionString} is an empty string, the {@code connectionString}
      * secret is invalid, or the HMAC-SHA256 MAC algorithm cannot be instantiated.
+     * @throws IllegalArgumentException if {@code tokenCredential} is not null. App Configuration builder support single
+     * authentication per builder instance.
      */
     @Override
     public ConfigurationClientBuilder connectionString(String connectionString) {
-        Objects.requireNonNull(connectionString, "'connectionString' cannot be null.");
+        if (this.tokenCredential != null) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException(
```

Comment:
This validation should happen in the build() method. User can set/reset this property but when they are attempting to build the client, there should be just one auth type.

Reply:
How do you reset? Null isn’t allowed.

Reply:
@srnagar We are also create a ConfigurationClientCredentials object by giving the connectionString, which null is not allowed. And endpoint updates based on the ConfigurationClientCredentials

Reply:
I can move those eargerly null/empty connectionstring validation to the build() if needed.